### ISSUE TITLE
[Question] Is it possible to use Storybook without transpilation, writing stories using pure ES6?

### ISSUE THREAD

**Issue by swergas** [storybookjs/storybook#12826]
Hi, I just have a general question about usage of Storybook. Is it possible to use Storybook without babel, writing stories using pure ES6? I would to use it in a project which does not use transpilation (React components written in pure ES6).

**Comment by shilman** [comment]
There's a prebuilt version of Storybook that supports this 

https://github.com/open-wc/open-wc/tree/master/packages/demoing-storybook

I'm not sure how to achieve this in standard Storybook cc @ndelangen

**Comment by ndelangen** [comment]
You can change the webpack config to what you see fit, you can you remove the babel-loader and any other loader you don't want.

If you don't want your stories and components to be passed through babel, it could be "as simple as":

```js
// main.js ( pseudo-code )

module.exports = {
  webpackFinal: async(config)=> {
    const filesNotPassedThroughBabelRegex = /some regex that excludes the files you don't want to pass through babel/
    config.module.rules[0].exclude.push(filesNotPassedThroughBabelRegex)
  }
}

```

**Comment by virtuoushub** [comment]
> There's a prebuilt version of Storybook that supports this
> 
> https://github.com/open-wc/open-wc/tree/master/packages/demoing-storybook
> 
> I'm not sure how to achieve this in standard Storybook cc @ndelangen


For anyone coming here in 2022, I found a prebuilt version of Storybook over at https://github.com/modernweb-dev/storybook-prebuilt

**Comment by valentinpalkovic** [comment]
Yes — in modern Storybook (7+), Vite-based frameworks like `@storybook/react-vite` use esbuild and don't require Babel. For webpack5, you can opt into SWC via `@storybook/addon-webpack5-compiler-swc`. The original question is fully resolved; closing as answered.
